### PR TITLE
Ble advertiser

### PR DESCRIPTION
**Pull request by shna6159**
Merging in the Proof of concept code in with main

### CODE REVIEW

**Review by AeroAtCU — APPROVED** [review]
Looks good

File: src/main.c [hunk 1]
```diff
@@ -92,21 +80,37 @@
...
+// #define TX_POWER_LEVEL 0
+// #define TX_POWER_LEVEL 4
+#define TX_POWER_LEVEL 8
+
+// #define BUTTON_DETECTION_DELAY          APP_TIMER_TICKS(50)                     /**< Delay from a GPIOTE event until a button is reported as pushed (in number of timer ticks). */
+
+// #define DEAD_BEEF                       0xDEADBEEF                              /**< Value used as error code on stack dump, can be used to identify stack location on stack unwind. */
+
+#define UUID_BASE {0x23, 0xD1, 0xBC, 0xEA, 0x5F, 0x78, 0x23, 0x15, 0xDE, 0xEF, 0x12, 0x12, 0x00, 0x00, 0x00, 0x00}
+#define UUID_SERVICE 0x1234
```
Comment (AeroAtCU) [inline src/main.c:93]: Should this have the same UUID as UUID_VOLTAGE_CHAR on the next line?

Comment (shna6159) [reply src/main.c:93]: The UUIDs are not delimiters, they are used to define the ID of the service and the characteristics. So, I randomly picked the hex values for the UUIDs. I am pretty sure it has no effect on which value I choose for the UUID. But I can change it to be unique just to be sure.

**Review by harrelljoseph91 — APPROVED** [review]
Looks pretty good to me

File: src/main.c [hunk 2]
```diff
@@ -355,21 +380,24 @@ static void conn_params_init(void)
...
 
 
 /**@brief Function for starting advertising.
  */
-static void advertising_start(void)
+static void ble_advertising_start(void)
 {
-    ret_code_t           err_code;
-
-    err_code = sd_ble_gap_adv_start(m_adv_handle, APP_BLE_CONN_CFG_TAG);
```
Comment (evan-meany) [inline src/main.c:369]: I think we should discuss adding error logging for future implementations.

**Review by evan-meany — COMMENTED** [review]
Looks good, good work Yash. I think now we should meet to discuss the best way to split up the remaining code so we can all contribute to getting everything working.

**Pull request merged**